docs/mkdocs/docs/features/namespace.md

```diff
@@ -0,0 +1,52 @@
+# `nlohmann` Namespace
+
+The 3.11.0 release introduced an inline namespace to allow different parts of a codebase to use different versions of
+the library as long as they never exchange instances of library types.
+
+In the 3.11.2 release, that namespace was split into the tag namespace – encoding the state of ABI-affecting macros –
+and the version namespace – encoding the version of the library.
+
+The full default namespace name is derived as follows:
+
+- The root namespace is always `nlohmann`.
+- The tag namespace starts with `nlohmann_json` and is followed by serveral optional ABI tags according to the value of
+  these ABI-affecting macros, in order:
+    - [`JSON_DIAGNOSTICS`](../api/macros/json_diagnostics.md) defined non-zero appends `_diag`.
+    - [`JSON_USE_LEGACY_DISCARDED_VALUE_COMPARISON`](../api/macros/json_use_legacy_discarded_value_comparison.md)
+      defined non-zero appends `_ldvcmp`.
+- The version namespace consists of the prefix `v` followed by the 3 components of the version number separated by
+  underscores.
+
+For example, the namespace name for version 3.11.2 with `JSON_DIAGNOSTICS` defined to `1`, is:
+
+```cpp
+nlohmann::nlohmann_json_diag::v3_11_2
+```
+
+## Disabling the version namespace
```

Comment:
I had another thought on the documentation.

The current documentation describes what the namespace is and how to manipulate the default behavior. Maybe it should be structured as follows:

- describe that starting in 3.11.0 we changed the namespace from `nlohmann` to a more elaborate version
- describe why this is a benefit - maybe by explicitly listing the incompatibilities that were possible before, but would not have been caught before; maybe also mention `JSON_SKIP_LIBRARY_VERSION_CHECK`
- describe how the namespace is structured
- describe the 3 different styles the namespace can be configured:
	- default - note: this is fine until we have a better means to describe ABI compatibility (right?)
	- no version part - note: this is fine if the library user wants to take the risk of mixing libraries with different versions, but with same ABI tags
	- `nlohmann` - note: use at your own risk

What do you think?

Reply:
I already had planned on addressing this, so thanks for the outline.

Reply:
I didn't quite adopt the outline and made a note to take another look and incorporate more of your ideas.

For now, I think what I have is as good as it's going to get in 2 passes.